factory.go

```diff
@@ -246,3 +248,52 @@ func (p *FilecoinParser) ParseGenesis(genesis *types.GenesisBalances, genesisTip
 
 	return genesisTxs, addresses
 }
+
+func (p *FilecoinParser) ParseGenesisMultisig(ctx context.Context, genesis *types.GenesisBalances, genesisTipset *types.ExtendedTipSet) ([]*types.MultisigInfo, error) {
+	var multisigInfos []*types.MultisigInfo
+	for _, actor := range genesis.Actors.All {
+		addrStr := actor.Key
+		// parse address
+		addr, err := address.NewFromString(addrStr)
+		if err != nil {
+			p.logger.Sugar().Errorf("could not parse address: %s. err: %s", addrStr, err)
+			continue
+		}
+
+		// get actor name from address
+		actorName, err := p.Helper.GetActorNameFromAddress(addr, int64(parser.GenesisHeight), genesisTipset.Key())
+		if err != nil {
+			p.logger.Sugar().Errorf("could not get actor name from address: %s. err: %s", addrStr, err)
+			continue
+		}
+
+		// check if the address is a multisig address
+		if !strings.EqualFold(actorName, manifest.MultisigKey) {
+			continue
+		}
+
+		api := p.Helper.GetFilecoinNodeClient()
+		metadata, err := multisigTools.GenerateGenesisMultisigData(ctx, api, addr, genesisTipset)
+		if err != nil {
+			return nil, fmt.Errorf("multisigTools.GenerateGenesisMultisigData(%s): %s", addrStr, err)
+		}
+
+		metadataJson, err := json.Marshal(metadata)
+		if err != nil {
+			return nil, fmt.Errorf("json.Marshal(): %s", err)
+		}
+
+		// there are no transactions for the multisig addresses in the genesis block
+		multisigInfo := &types.MultisigInfo{
+			ID:              tools.BuildId(genesisTipset.GetCidString(), addrStr, fmt.Sprint(parser.GenesisHeight), "", parser.TxTypeGenesis),
+			MultisigAddress: addrStr,
+			Height:          parser.GenesisHeight,
+			Signer:          addrStr,
```

Comment:
In this case maybe we don't have Signer, because is genesis. Am I wrong? @emmanuelm41

Reply:
Yep, you are right. The Signer in this case is nobody, as it is genesis